apps/dash-daq-tektronix350/app_mock.py

```diff
@@ -0,0 +1,597 @@
+    [Input("toggleTheme", "value"), Input("color-picker", "value")],
+    [State("control-inputs", "data"), State("tabs", "value")],
+)
+def turn_dark(turn_dark, color_pick, cur_inputs, cur_tab_value):
+    theme.update(dark=turn_dark)
+
+    if color_pick is not None:
+        theme.update(primary=color_pick["hex"])
+
+    return DarkThemeProvider(
+        theme=theme,
+        children=[
+            power_setting_div(cur_inputs, cur_tab_value),
+            function_setting_div(cur_inputs, cur_tab_value),
+        ],
+    )
+
+
+# Update colors upon color-picker changes
+@app.callback(
+    [
+        Output("power-title", "style"),
+        Output("function-title", "style"),
+        Output("graph-title", "style"),
+        Output("div-graph-info", "style"),
+        Output("tabs", "color"),
+    ],
+    [Input("color-picker", "value")],
+)
+def color_update(color):
+    return (
+        list({"color": color["hex"]} for _ in range(3))
+        + [{"border": ("1px solid " + color["hex"]), "color": "inherit"}]
+        + [{"border": color["hex"]}]
+    )
+
+
+# Callback updating backgrounds
+@app.callback(
+    [
+        Output("main-page", "className"),
+        Output("left-panel", "className"),
+        Output("card-right-panel-info", "className"),
+        Output("card-graph", "className"),
+    ],
+    [Input("toggleTheme", "value")],
+)
+def update_background(turn_dark):
+
+    if turn_dark:
+        return ["dark-main-page", "dark-card", "dark-card", "dark-card"]
+    else:
+        return ["light-main-page", "light-card", "light-card", "light-card"]
+
+
+# Callbacks for knob inputs
+@app.callback(Output("frequency-display", "value"), [Input("frequency-input", "value")])
+def update_frequency_display(value):
+    return value
+
+
+@app.callback(Output("amplitude-display", "value"), [Input("amplitude-input", "value")])
+def update_amplitude_display(value):
+    return value
+
+
+@app.callback(Output("offset-display", "value"), [Input("offset-input", "value")])
+def update_offset_display(value):
+    return value
+
+
+# Callback for adding tabs
+@app.callback(
+    [Output("tabs", "children"), Output("tabs", "value")],
+    [Input("new-tab", "n_clicks")],
+    [State("control-inputs", "data"), State("color-picker", "value")],
+)
+def update_total_tab_number(n_clicks, cur_inputs, color):
+    return (
+        list(
+            dcc.Tab(
+                label="Run #{}".format(i),
+                value="{}".format(i),
+                selected_style={
+                    "color": color["hex"],
+                    "backgroundColor": "transparent",
+                    "border": "none",
+                },
+                style={
+                    "color": "black",
+                    "backgroundColor": "transparent",
+                    "border": "none",
+                },
+            )
+            for i in range(1, len(cur_inputs) + 2)
+        ),
+        str(len(cur_inputs) + 1),
+    )
+
```

Comment:
What is the purpose of the mock app file?

Reply:
For the `daq` apps there are two files: the mock and the actual app.

The `app_mock.py` file is the one being shown on gallery. The `app.py` is supposed to be used with an actual function generator machine so you don't have to worry about that one.